codice/ddf
# DDF-4811 Match the coordinate order of each WFS 1.1.0 query to that of its source

#### What does this PR do?
Updates the WFS 1.1.0 `FilterDelegate` to use the configured coordinate order of its corresponding source to determine the coordinate order to use when it constructs the query.

#### Who is reviewing it?
@Corey-Collins 
@samuelechu 
@leo-sakh 
@glenhein 

#### Select relevant component teams: 
@codice/io 
@codice/ogc 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@jlcsmith

#### How should this be tested?
1. Run `log:set info org.apache.cxf` and `log:set warn org.apache.camel`.
2. Create a WFS 1.1.0 source with the URL http://services.azgs.az.gov/ArcGIS/services/aasggeothermal/AZWellLogs/MapServer/WFSServer
3. Run various `anyGeo` queries against the source, using `Line`, `Polygon`, `Point-Radius`, and `Bounding Box`.
a. When you run a query there will be a `GetFeature` XML blob printed in the logs that contains the geospatial query. Use this to verify each query's correctness.
b. For each query, run it as both `Lat/Lon` and `Lon/Lat` and verify the outgoing `GetFeature` XML has the right coordinate order in both cases. Check the `Coordinate Order` in your WFS source's configuration, then run your query and verify that the coordinate order in the `GetFeature` matches. Change the `Coordinate Order` in the source configuration, rerun the same query, and verify the coordinate order in the `GetFeature` changed appropriately.

#### What are the relevant tickets?
Fixes: #4811 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
